### PR TITLE
Adjust AUTHORS to reflect current state of the project

### PR DESCRIPTION
**Pull request by hobu**
I've updated AUTHORS to more accurately reflect the state of the project and give people a pointer to RFC 1 if they want more information.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
+................................................................................
 
-Contributors:
 Brent Fraser <bfraser@geoanalytic.com>
 Chris Stuber <imap@chesapeake.net>
 Craig Bruce <cbruce@cubewerx.com>
 Victor Osipkov <vctos@email.com>
 Andrea Antonello <andrea.antonello@hydrologis.com>
-Charles Karney <charles.karney@sri.com>
-Karsten Engsager
```
Comment (kbevers) [inline AUTHORS:15]: Why is Karsten left out?

**Review by kbevers — COMMENTED** [review]
Thanks for updating this, Howard.

I wonder if we should remove the contributor list altogether? We are not updating it any more since git keeps track of who has contributed code. I haven't checked but I would assume that the remaining people in the contributor list has their name written somewhere near the code they have added to the project. I know at least that is true for Karsten Engsager and Knud Poder. Instead we can put a link to https://github.com/OSGeo/proj.4/graphs/contributors

File: AUTHORS (hunk 1 above)
Comment (hobu) [reply AUTHORS:15]: typo

**Comment by hobu** [comment]
I wasn't quite sure what we wanted to do for AUTHORS so I proposed this to start the conversation. I agree that git takes care of the contributor tracking, but it is nice to have a list that gets pushed out with the distros.

**Comment by kbevers** [comment]
I mostly think that it is a relic from the past. I don't mind keeping it around but I think that it gives way too much credit to a few drive by contributors from +15 years ago. Having past and current maintainers and PSC members as well as a link to the GitHub contributor list strikes a good balance in my opinion.

**Comment by kbevers** [comment]
I can't find these guys anywhere in the source code:

```
Brent Fraser <bfraser@geoanalytic.com>
Chris Stuber <imap@chesapeake.net>	Chris Stuber <imap@chesapeake.net>
Craig Bruce <cbruce@cubewerx.com>	Craig Bruce <cbruce@cubewerx.com>
Victor Osipkov <vctos@email.com>	Victor Osipkov <vctos@email.com>
```

It would be best to see if we can figure out what they originally contributed and add their credit to the relevant source files.

**Comment by rouault** [comment]
Some archeology... This is difficult since before git, contributions from casual contributors were committed by someone else who didn't necessarily mentionned original author.
Brent Fraser  and Chris Stuber were added in AUTHORS per 129523507cd1a0faaa6d1d9e70089d2f3c56422f of March 2000 which is the 16th commit since the beginning of PROJ CVS history...  Chris Stuber only explicit mention in commit history is 8bd626e8dc73ff5ace8523ed3601b1d0bdb03452 a few revision before for a trivial change.
Cannot find any trace in commit messages of Brent Fraser although he has posted a few times on the mailing list over years.
Craig Bruce only explicit contribution in commit messages are 067352cad73074fe83ea5a779de03e68d36e5783 in April 2000 and d503ded5a03e626b093b21f2e6adb842560a4260 in Nov 2000
Victor Osipkov was added in AUTHORS per c92c5d8e556365e3b01abb72d340ed91576703a6 in April 2001. Couldn't find for which commit(s) he deserved credit
So overall, from the above, I don't feel their contributions deserves more credit than the one of more recent contributors which are not mentionned in AUTHORS.
Or perhaps have a section for contributors of the git era where we point to github contributor list as @kbevers suggested, and keep a section "Past contributors (before git)" with those few names.

**Comment by busstoptaktik** [comment]
Having recently had a reviewer of an academic paper getting seriously confused over our currently suggested reference “PROJ contributors (2018)”, I think we should not only retain the AUTHORS file, but also do more in terms of making PROJ properly attributable.

This may be a matter of attempting to get a paper accepted for the Journal of Open Source Software (JOSS), or somesuch, for each major version of PROJ.

For versions 3, 4 and 4.3(?) there are USGS reports by Gerald available as reference targets. For version 5, there are  **no evident references**, although two conference papers due to @kbevers and myself are available as stop gaps if needed. And for version 6 on, we could attempt to go the JOSS route.

Hence, a **complete historical** reference to PROJ could read something like “Evenden (1983), Evenden (1990), Evenden (1994), Evers & Knudsen (2017), Evers et al (2019)”  (where the "Evers & Knudsen 2017 part should be considered left out, since it will give disproportional credit to myself and @kbevers - but we lack a proper 5.0 reference).

Whereas a **sufficient** in-text reference would simply read “Evers et al (2019)” (where I suppose the release manager would be the primary author of a JOSS paper).

With a JOSS paper published, the relevant PROJ version would be uniquely identified with a DOI, and indexed via crossref.org, hence being much more (academically) visible, making it attractive for academic contributions, and probably also more visible in the society at large (since “who does *not* use PROJ - but who knows how to reference PROJ”).

But in general I think we should keep the AUTHORS file - and probably supplement with other relevant metadata from the Dublin Core set. Librarians and information scientists are conservative, and they are for a reason: They build systems and concepts (such as the Dublin Core) for multi millenia information conservation, so they tend to prefer proven ways.

@kbevers’s concerns about giving too much eternal credit to distant drive-by contributors could be solved by splitting the file in sections related to each (major?) release, in reverse chronological order, such that Gerald Evenden is attributed as sole author of releases up to and including version 3, i.e. “Prior to version 4”, then for version 4 all contributors between 4 and 4.9.3 would be listed, an so on for version 5, 6, and 7.

Hence Gerald and Frank will drop off the list for version 5. This may, taking their historical roles into consideration, be unacceptable, but it also resolves the problem of eternal attribution of single patch contributors.

**Comment by busstoptaktik** [comment]
> Some archeology... This is difficult since before git, contributions from casual contributors were committed by someone else who didn't necessarily mentionned original author.

As per personal experience: In the spring of 1999, I had material accepted which I had sent directly by email to Frank. He left a note with my initials in the source code to proj.c, where it remained until I removed it myself, and at the same occasion, added my name to AUTHORS, more than 15 years later.

So historically, official authorship has required somewhat more than just my repairing a bug in the proj I/O on Windows platforms. How much may differ :-)

**Comment by hobu** [comment]
https://zenodo.org/ offers DOIs for software. We started using this for PDAL releases, and maybe it would be applicable for PROJ.

> Hence Gerald and Frank will drop off the list for version 5. This may, taking their historical roles into consideration, be unacceptable, but it also resolves the problem of external attribution of single patch contributors.

This is completely unacceptable to me. The architecture that both laid down did not change in version 5. They're authors.

I didn't mean to open up a can of worms here. AUTHORS was out of date in relation to the current activity. It is an artifact that is pushed out and serialized with the release. IMO, if you had something merged, you're a contributor, drive-by or not. AUTHORS is the list of people needed to be contacted in the event of a license change (this is never happening for PROJ). It isn't an academic authorship with ordering determining some kind of rank. 

Maybe we should simply back this off to PSC membership, point to the website's topic, and credit Gerald memorially at the top.

**Comment by kbevers** [comment]
> Maybe we should simply back this off to PSC membership, point to the website's topic, and credit Gerald memorially at the top.

This is my preferred option.

I would also like to see the possibilities for academic referencing improved but that is outside the scope of this PR in my opinion.

**Comment by busstoptaktik** [comment]
> This is completely unacceptable to me. The architecture that both laid down did not change in version 5. They're authors.

@hobu: You know very well that I am of the same opinion. I was just trying to demonstrate some obvious drawbacks to discerning between "levels of involvement".

Your remark that AUTHORS track the people who should be contacted in case of a license change is just another argument for keeping AUTHORS more or less as-is, although I think Gerald and Frank deserve special prominent mentioning.

Perhaps something like what has been used for gnuplot for some decades now: Attributing the original authors, then the maintainer/release manager, then major contributors to the current release.

Williams & Kelley have not contributed new material for (probably) decades, but the front page of the manual announces that:

> gnuplot 5.2 - An Interactive Plotting Program
>
> Thomas Williams & Colin Kelley
>
> Version 5.2 organized by: Ethan A Merritt and many others
>
> Major contributors (alphabetic order): Christoph Bersch, Hans-Bernhard Bröker, John Campbell, Robert Cunningham, David Denholm, Gershon Elber ... etc etc.
>

So read "Gerald I. Evenden & Frank Warmerdam" for "Thomas Williams and Colin Kelley", add a section of "Past contributors" for contributors to past major releases, and I think we have something quite workable, which fits into the not entirely identical opinions stated by @hobu and @kbevers in this thread.

@hobu: My comments in this thread are not intended as review or other kind of comment to your PR (which I think should just be merged), but we needed some place to start the discussion, and your PR provided a nice occasion (as, according to one of your first remarks, was also your intention).

We can probably continue this discussion in the PSC: It is evident that we need some academically viable way of referring to PROJ.

**Comment by hobu** [comment]
I removed the contributors list and simply pointed at https://github.com/OSGeo/proj.4/graphs/contributors

**Comment by kbevers** [comment]
Thanks Howard!

**Comment by busstoptaktik** [comment]
@hobu & @kbevers, while I like the nice structure of the new AUTHORS file, I find the removal of the contributors' names problematic: Attribution is *part* of the work, not something external to it.

**Just providing a link would be like Amazon removing the title pages of books distributed and adding a notice that "the author information is printed on your invoice".**

Additionally, the link eliminates attribution of the work by Karsten Engsager & Knud Poder - and I really think etmerc & horner is a more important contribution than a handful of minor corrections now (rightfully) attributed.

As I'm neither a lawyer nor a librarian, I am not able to find the proper formal terms for my objection. I have, however, spent sufficient time with copyright lawyers (as expert witness in a prominent copyright case) and with librarians (as a library assistant during my study years), to find the removal of in-publication author attribution deeply concerning.

I find it seriously important to reinstall the full list of contributors before release of 6.0. We may be able to find the proper judicial and information management terms for removal later, but right now we have none. To the contrary we have indications that this is wrong.

**Comment by kbevers** [comment]
> @hobu & @kbevers, while I like the nice structure of the new AUTHORS file, I find the removal of the contributors' names problematic: Attribution is _part_ of the work, not something external to it.
>
> **Just providing a link would be like Amazon removing the title pages of books distributed and adding a notice that "the author information is printed on your invoice".**

I see `AUTHORS` as a list of primary contributors. A primary contributor to me is someone who has either made significant code contributions or have played central parts in the management of the PROJ project. This aligns very well with what was submitted in this PR.

For every release I make a list of contributors to that release in `NEWS`. I do that because I too think that attribution is important. So I think your Amazon allegory is unfair to what we actually do. The alternative, at least how I understand your proposol/requst, is that we update `AUTHORS` everytime a new contributor shows up. We could do that but I fear that is a system with a high likelyhood of falling apart.

> Additionally, the link eliminates attribution of the work by Karsten Engsager & Knud Poder - and I really think etmerc & horner is a more important contribution than a handful of minor corrections now (rightfully) attributed.

Engsager and Poder are attributed plenty in "the work":

```
grep -i -R --exclude-dir=build -e Engsager -e Poder
docs/source/operations/projections/tmerc.rst:The default algorithm is given by Poder and Engsager in :cite:`Poder1998`
docs/source/references.bib:@techreport{Poder1998,
docs/source/references.bib:  author       = {Knud Poder and Karsten Engsager},
examples/s45b.pol:# fwd-inv ombyttet ifht original Poder/Engsager-kode
src/projections/tmerc.cpp:* Evenden/John Snyder origin and one of Knud Poder/Karsten Engsager origin. The
src/projections/tmerc.cpp:* and the Poder/Engsager algorithm is a slightly slower, but more accurate
src/projections/tmerc.cpp:// Written by: Knud Poder and Karsten Engsager
src/projections/tmerc.cpp:    /* PROJ_ETMERC_ORDER = 6th degree : Engsager and Poder: ICC2007 */
src/projections/tmerc.cpp:/* UTM uses the Poder/Engsager implementation for the underlying projection      */
src/transformations/helmert.cpp:        For comparison: Description from Engsager/Poder implementation
src/transformations/horner.cpp:    2D Horner's Scheme polynomial evaluation routine by Knud Poder and
src/transformations/horner.cpp:    Karsten Engsager, originating in the vivid geodetic environment at
src/transformations/horner.cpp:    The original Poder/Engsager gen_pol implementation (where
src/transformations/horner.cpp:    The Poder coding sound, as classic 1960s as Phil Spector's Wall
src/transformations/horner.cpp: famous "Poder dual autocheck" in all its enigmatic elegance.
src/transformations/horner.cpp: The material included here was written by Knud Poder, starting
src/transformations/horner.cpp: around 1960, and Karsten Engsager, starting around 1970. It was
src/transformations/horner.cpp: * Copyright (c) 2016, SDFE http://www.sdfe.dk / Thomas Knudsen / Karsten Engsager
src/transformations/horner.cpp:A reimplementation of the classic Engsager/Poder 2D Horner polynomial
src/transformations/horner.cpp:This version omits the inimitable Poder "dual autocheck"-machinery,
src/transformations/horner.cpp:typical when Knud Poder invented the dual autocheck method,
src/transformations/horner.cpp:    /* These variable names follow the Engsager/Poder  implementation */
src/transformations/horner.cpp:    /* The melody of this block is straight out of the great Engsager/Poder songbook */
src/transformations/horner.cpp:A reimplementation of a classic Engsager/Poder Horner complex
src/transformations/horner.cpp:    /* These variable names follow the Engsager/Poder  implementation */
```

Additionally, as far as I am aware the contributions by Engsager and Poder have all been made by you, somewhat on their behalf. At least the Horner code. I don't know if "second-hand contributions" like should be treated differently or not.

For most of the pre-svn/git days attributions to the original author is listed in the code. If we were to maintain a complete list of contributors in `AUTHORS` we would have to go through the code and find all of them. I don't think that is something anyone of us want to do. In my opinion that is attribution enough, and it still comes as part of the package that we distribute.

So, what we have now is

- A list of primary contributors in `AUTHORS` 
- Credit to all contributors in a specific release in `NEWS`
- Credit to contributions from pre-svn/git days locally to the contributed code
- Contributions tracked in the git repository and the contributor list on GitHub (which is also linked in `AUTHORS`)

That leaves us with the few names that has been removed in this PR that isn't attributed anywhere else. Since it has been impossible to figure out what those few persons has contributed I regard it as fairly likely that their code has been removed, or at least been rewritten significantly. Ideally we would be able to determine their contributions and add a note about the origin of that part of the code but I fear that will be impossible. So while this solutions is not perfect it at least is doing as much as possible while also clearing up things for the future and making the `AUTHOR` file easier to maintain (because it is now clear what should be in it).

**Comment by kbevers** [comment]
Doing a bit more digging, I can find specific mentions of contributions of Craig Bruce and Andrea Antonello. I can't find any trace of what and when Brent Fraser, Chris Stuber and Victor Osipkov contributed:

```
>grep -i -R --exclude-dir=build -e "Brent Fraser"

>grep -i -R --exclude-dir=build -e "Chris Stuber"

>grep -i -R --exclude-dir=build -e "Craig Bruce"
ChangeLog:      * Patch from Craig Bruce to adjlon.c to avoid wrong results,
ChangeLog:      * Patch from Craig Bruce (cbruce@cubewerx.com) for PJ_ortho.c

>grep -i -R --exclude-dir=build -e "Victor Osipkov"

>grep -i -R --exclude-dir=build -e "Andrea Antonello"
ChangeLog:      * Added jniwrap subtree (actually Andrea Antonello).
jniwrap/README.md:* Andrea Antonello (andrea.antonello@hydrologis.com)
src/jniproj.cpp: * Copyright (c) 2005, Andrea Antonello
```

Andre Antonello is already credited in the JNI files. We should mention Craig Bruce in both `src/projections/ortho.cpp` and `src/adjlon.cpp`. The three remaining will unfortunately end up as collateral damage, as I can't find any reference to what they may have done anywhere. Neither in the git repository or on the mailing list.

**Comment by rouault** [comment]
> I can't find any trace of what and when Brent Fraser, Chris Stuber and Victor Osipkov contributed:

See https://github.com/OSGeo/proj.4/pull/1287#issuecomment-466695992 which mostly concurs with your analysis with the extra info "Chris Stuber only explicit mention in commit history is [8bd626e](https://github.com/OSGeo/proj.4/commit/8bd626e8dc73ff5ace8523ed3601b1d0bdb03452) a few revision before for a trivial change". But I now realize I removed biveval.c[pp] last week...

**Comment by kbevers** [comment]
Sorry, I overlooked that you indeed had found trace of Chris Stuber. Good detective work! But yeah, It's going to be difficult to credit him for code that isn't part of PROJ anymore. So we're basically down to two persons that we can't find any contributions from.

**Comment by hobu** [comment]
gah, what a swamp 😄 I'm deferring the decisions on how to approach this to the PSC chair.

**Comment by busstoptaktik** [comment]
@kbevers opined:

> I see `AUTHORS` as a list of primary contributors. A primary contributor to me is someone who has either made significant code contributions or have played central parts in the management of the PROJ project. This aligns very well with what was submitted in this PR.

@kbevers, this begins to make sense to me: Contribution is not the same as authorship (an editor or copywriter, e.g., is also contributing, but not authoring).

Using terms from scientific publishing, we could discern between people in the list of authors (which may include some not having contributed any text, but perhaps data, analysis or other important work), and people attributed in the "Acknowledgements" section (which may very well have provided actual text, say from an e-mail conversation clarifying a concept, or summarizing related work), but do not qualify for authorship.

So what this PR have accomplished is, a.o., to make it more clear what it requires to qualify as author (belonging in the AUTHORS list). Remaining committers are considered contributors (and attributed not in the "Acknowledgements" section, but in the NEWS file).

This is, however, not in line with the @hobu comment that AUTHORS lists those whom should be contacted in case of license changes. Since this is unlikely to happen, and since the NEWS file is accumulative (at least back to version 4.4.6), it should be possible to track down all relevant contributors, though. And, if necessary, we could probably extend at the bottom with an "old news" section tracking relevant "old news" as it gets uncovered, by @rouault and his world famous team of PROJ archaeologists :-)

> Additionally, as far as I am aware the contributions by Engsager and Poder have all been made by you, somewhat on their behalf. At least the Horner code. I don't know if "second-hand contributions" like should be treated differently or not.

This is correct as for `horner`, but a much more complicated story for `etmerc`. 

You can probably look forward to enjoy the details, when I retire in 15-20 years or so, and get time to reiterate in my (supposedly) upcomming blog series "Particularly interesting tidbits from the obscure corners of the history of computational geodesy". Currently, the details are all in my 2008 email archive, and involves convolved iteration between myself, Karsten Engsager, Frank Warmerdam, Gerald I. Evenden, and (offline, through Karsten) Knud Poder.

Later improvements are due to @cffk - whom, I believe, has also corresponded directly with Knud Poder, but I'm not sure whether this is related to his geodesic work only.

**Comment by kbevers** [comment]
> I'm deferring the decisions on how to approach this to the PSC chair.

I stand by what I wrote above in https://github.com/OSGeo/proj.4/pull/1287#issuecomment-466955931. I believe we are on safe ground with regards to attributions. I will change one thing about my statement though: After having looked over the two patches by Craig Bruce, I don't think it is necessary to add his name to the files that he made changes to. In both cases we are talking very minor changes, and for the `adjlon`patch, it has even been rewritten partly afterwards. He is mentioned in the Changelog though, so his contributions are not completely overlooked.

In conclusion, after the various levels of attributions has been clarified I don't think that further change to `AUTHORS` is needed at this time.
